### PR TITLE
Fix Maryland Earned Income Tax Credit (EITC) 

### LINKED ISSUE #4247: Fix Maryland Earned Income Tax Credit (EITC)
state earned income credit: https://www.marylandtaxes.gov/forms/21_forms/Resident_Booklet.pdf#page=19
refundable earned income credit: https://www.marylandtaxes.gov/forms/21_forms/Resident_Booklet.pdf#page=23

Legal code: https://casetext.com/statute/code-of-maryland/article-tax-general/title-10-income-tax/subtitle-7-income-tax-credits/section-10-704-effective-until-6302023-for-earned-income

### PR DESCRIPTION
**Pull request by Shinuing**
Fix #4247 

Main modifications: 
1. Separate non-refundable and refundable credit 
2. Rename “single” to “unmarried” as single here means filing status as single, hoh or widow. 
3. Added unmarried childless non-refundable eitc

### CODE REVIEW

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: does this fix the integration test from the discussion?

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/married_or_has_child/match.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-description: Maryland matches this percent of the federal Earned Income Tax Credit as a non-refundable credit for filers who are not single and childless.
+description: Maryland matches this percent of the federal Earned Income Tax Credit for individuals with qualifying child or married couple filing jointly or seperately with or without qualifying child.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/married_or_has_child/match.yaml:1]:
```suggestion
description: Maryland matches this percent of the federal Earned Income Tax Credit for individuals with qualifying child or married couples filing jointly or separately with or without a qualifying child.
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/unmarried_childless/max_amount.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-description: Maryland provides up to this amount for its Earned Income Tax Credit to single childless filers.
+description: Maryland provides up to this amount for its Earned Income Tax Credit to individuals without qualifying child.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/unmarried_childless/max_amount.yaml:1]:
```suggestion
description: Maryland provides up to this amount for its Earned Income Tax Credit to individuals without a qualifying child. 
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/unmarried_childless/percent_match.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-description: Maryland provides up to this percent of the federal Earned Income Tax Credit to single childless filers.
+description: Maryland provides up to this percent of the federal Earned Income Tax Credit to individuals without qualifying child.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/unmarried_childless/percent_match.yaml:1]:
```suggestion
description: Maryland matches this percentage of the federal Earned Income Tax Credit to individuals without a qualifying child. 
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/unmarried_childless/percent_match.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/unmarried_childless/percent_match.yaml:1]: can you rename the file to match?

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/md_qualifies_for_unmarried_childless_eitc.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
 from policyengine_us.model_api import *
 
 
-class md_qualifies_for_single_childless_eitc(Variable):
+class md_qualifies_for_unmarried_childless_eitc(Variable):
     value_type = bool
     entity = TaxUnit
-    label = "Qualifies for the MD single childless EITC"
+    label = "Qualifies for the MD unmarried childless EITC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/md_qualifies_for_unmarried_childless_eitc.py:7]:
```suggestion
    label = "Qualifies for the Maryland unmarried childless EITC"
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class md_married_or_has_child_non_refundable_eitc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "MD non-refundable EITC for filers who are married or have qualifying child"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py:7]:
```suggestion
    label = "Maryland non-refundable EITC for filers who are married or have qualifying child"
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    unit = USD
+    definition_period = YEAR
+    reference = "https://casetext.com/statute/code-of-maryland/article-tax-general/title-10-income-tax/subtitle-7-income-tax-credits/section-10-704-effective-until-6302023-for-earned-income"  # (c)(1)
+    defined_for = StateCode.MD
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py:11]:
```suggestion
    defined_for = "md_qualifies_for_unmarried_childless_eitc"
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        federal_eitc = tax_unit("eitc", period)
+
+        p = parameters(
+            period
+        ).gov.states.md.tax.income.credits.eitc.non_refundable.married_or_has_child
+        # Limited to filers who are married or have child
+        married_or_has_child = ~tax_unit(
+            "md_qualifies_for_unmarried_childless_eitc", period
+        )
+        amount = federal_eitc * p.match
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py:23]:
```suggestion
       return federal_eitc * p.match
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        married_or_has_child = ~tax_unit(
+            "md_qualifies_for_unmarried_childless_eitc", period
+        )
+
+        federal_eitc = tax_unit("eitc", period)
+        md_tax = tax_unit("md_income_tax_before_credits", period)
+        p = parameters(
+            period
+        ).gov.states.md.tax.income.credits.eitc.refundable.married_or_has_child
+        md_refundable_eitc_allowed = max_(federal_eitc * p.match - md_tax, 0)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py:24]: can you make a md_refudnable_eitc file? and use defined for here

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_unmarried_childless_refundable_eitc.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class md_unmarried_childless_refundable_eitc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "MD unmarried childless EITC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_unmarried_childless_refundable_eitc.py:7]:
```suggestion
    label = "Maryland unmarried childless EITC"
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (Shinuing) [reply changelog_entry.yaml:1]: The refundable eitc passed. 

As far as non-refundable one, I'm still confused. Let me know which one I should follow.

Based on the [tax form](https://www.marylandtaxes.gov/forms/21_forms/Resident_Booklet.pdf#page=19): non-refundable eitc = 530
Screen Shot 2024-03-21 at 11 42 59 AM

Based on the [legal code](https://casetext.com/statute/code-of-maryland/article-tax-general/title-10-income-tax/subtitle-7-income-tax-credits/section-10-704-for-earned-income): credit against the state income tax = 280.4 = TAXSIM result
Screen Shot 2024-03-21 at 11 48 14 AM

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py (hunk 6 above)
Comment (Shinuing) [reply policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py:11]: This is not defined for "md_qualifies_for_unmarried_childless_eitc" but the opposite one

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py (hunk 8 above)
Comment (Shinuing) [reply policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py:24]: can you be more clear? there's already a `md_refundable_eitc` file for the total refundable eitc

File: changelog_entry.yaml
Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: In this case I would follow the tax form as that is what the people will be actually filling out. 

@MaxGhenis FYI

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py (hunk 8 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py:24]: My mistake meant to say md_refundable_eitc_eligible - just as an eligibility file to separate it out in the calc

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_unmarried_childless_refundable_eitc.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class md_unmarried_childless_refundable_eitc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Maryland unmarried childless EITC"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_unmarried_childless_refundable_eitc.py:7]:
```suggestion
    label = "Maryland refundable unmarried childless EITC"
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    defined_for = StateCode.MD
+
+    def formula(tax_unit, period, parameters):
+        federal_eitc = tax_unit("eitc", period)
+
+        p = parameters(
+            period
+        ).gov.states.md.tax.income.credits.eitc.non_refundable.married_or_has_child
+        # Limited to filers who are married or have child
+        married_or_has_child = ~tax_unit(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/non_refundable/md_married_or_has_child_non_refundable_eitc.py:20]:
```suggestion
        does_not_qualify_for_unmarried_childless_eitc = ~tax_unit(
```

File: policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/credits/eitc/refundable/md_married_or_has_child_refundable_eitc.py:24]:
```suggestion
        total_refundable_eitc_amount = federal_eitc * p.match
        md_refundable_eitc_allowed = max_(total_refundable_eitc_amount - md_tax, 0)
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**